src/lib/cfa/chrony_conf.rb

```diff
@@ -0,0 +1,142 @@
+require "cfa/base_model"
+require "cfa/augeas_parser"
+require "cfa/matcher"
+require "cfa/placer"
+
+module CFA
+  # class representings /etc/chrony.conf file model. It provides helper to manipulate
+  # with file. It uses CFA framework and Augeas parser.
+  # @see http://www.rubydoc.info/github/config-files-api/config_files_api/CFA/BaseModel
+  # @see http://www.rubydoc.info/github/config-files-api/config_files_api/CFA/AugeasParser
+  #
+  class ChronyConf < ::CFA::BaseModel
+    PATH = "/etc/chrony.conf".freeze
+
+    def initialize(file_handler: nil)
+      super(CFA::AugeasParser.new("chrony.lns"), PATH, file_handler: file_handler)
+    end
+
+    # loads cfa model and ensure all collection keys have [] suffix
+    def load
+      super
+      fix_collection_names(data)
+    end
+
+    # options hash with key value, where value can be nil for keyword options
+    # or :default to use default options
+    def add_pool(address, options = :default)
+      options = default_pool_options if options == :default
+      # if there is already pool entry, place it after, if not, try use comment
+      existing_pools = pure_pools
+      if existing_pools.empty?
+        # for now first chrony have pools under comment mentioning pool.ntp.org
+        # so try to place it below
+        matcher = Matcher.new { |k, v| k.start_with?("#comment") && v =~ /www\.pool\.ntp\.org/ }
```

Comment:
so if user removes the comment we fail?

Reply:
no, AfterPlacer if matcher does not find entry, append it to end of file. So strategy is:

1. if there is any pool, place it afterwards
2. place after comment for pools
3. append to end of file

Reply:
no ... just add at the end of conf file (tested)

Reply:
I will document that behavior about where line is placed.